File: Example/ObjectC_Example/WI BLE Lib Sample objc/WI BLE Lib Sample objc/ViewController.m
//
//  ViewController.m
//  WI BLE Lib Sample objc
//
//  Created by ChangsoonKim on 11/06/2019.
//  Copyright © 2019 ZOYI Corporation. All rights reserved.
//

#import "ViewController.h"
#import <WIBLELib/WIBLELib-Swift.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [BleManager setDebugMode:false];
    
    _manager = [[BleManager alloc] initWithInitWithDelegate:self];
    [_tbMacs setText:@"f4fd2b10780a, f4fd2b2086fe"];
    [_tbRSSI setText:@"-70"];
}
    
- (void) showAlert:(NSString *)mac withRssi:(NSInteger)rssi {
    NSString *title = [NSString stringWithFormat:@"Find Mac %@ - %ld", mac, (long)rssi];

    UIAlertController* alert = [UIAlertController alertControllerWithTitle:title
                                message:@"Success"
                                preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:NULL];
    
    [alert addAction:defaultAction];
    [self presentViewController:alert animated:YES completion:nil];
}

- (void) startScanning {
    NSArray* macs = [[_tbMacs text] componentsSeparatedByString:@", "];
    
    if (macs != NULL &&
        ![_manager isScanning] &&
        [_manager isPowerOn] &&
        [_manager startScanWithMacsWithTargetMacs:macs])
    {
        _rssiThreashHold = [[_tbRSSI text] intValue];
        [_btnStart setTitle: @"Stop" forState: UIControlStateNormal];
        [_tbRSSI setUserInteractionEnabled:false];
        [_tbMacs setUserInteractionEnabled:false];
    }
}

- (void) stopScanning {
    NSLog(@"stop Scanning");
    [_manager stopScan];
    [_btnStart setTitle: @"Start" forState: UIControlStateNormal];
    [_tbRSSI setUserInteractionEnabled:true];
    [_tbMacs setUserInteractionEnabled:true];
}
    

- (IBAction)touchUpStartBtn:(id)sender {
    NSLog(@"clicked");
    if ([_manager isScanning]) {
        [self stopScanning];
        return;
    }
    [self startScanning];
}
    
- (void)didDiscoverMacWith:(NSString * _Nonnull)mac rssi:(NSInteger)rssi timestamp:(NSTimeInterval)timestamp {
    if (rssi > [self rssiThreashHold]) {
        NSLog(@"Find %@", mac);
        [self showAlert:mac withRssi:rssi];
        [self stopScanning];
    }    
}

@end

File: Example/Swift_Example/WI BLE Lib Sample/WI BLE Lib Sample/ViewController.swift
//
//  ViewController.swift
//  WI BLE Lib Sample
//
//  Created by ChangsoonKim on 10/06/2019.
//  Copyright © 2019 ZOYI Corporation. All rights reserved.
//

import UIKit
import WIBLELib

class ViewController: UIViewController, BleManagerDeleagate {
    var manager:BleManager!
    var rssiThreashHold:Int!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Do any additional setup after loading the view.
        BleManager.debugMode = false
        manager = BleManager(initWithDelegate: self)
        
        tbRSSI.text = "-70"
        tbTargetMacs.text = "f4fd2b10780a, f4fd2b2086fe"
        
    }

    @IBOutlet weak var btnStartStop: UIButton!
    @IBOutlet weak var tbRSSI: UITextField!
    @IBOutlet weak var tbTargetMacs: UITextField!
    
    func didDiscoverMac(with mac: String, rssi: Int, timestamp: TimeInterval) {
        if (rssi > rssiThreashHold) {
            print("Find!! \(mac)")
            self.showAlert(mac: mac, rssi: rssi)
            self.stopScanning()
        }
    }
    
    func showAlert(mac: String, rssi: Int) {
        let alert = UIAlertController(title: "Find Mac(\(mac) - \(rssi))", message: "Success", preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
        
        self.present(alert, animated: true)
    }
    
    func startScanning() {
        let targetMacs = tbTargetMacs.text?.components(separatedBy: ",").map{$0.trimmingCharacters(in: .whitespacesAndNewlines)}
        
        if (targetMacs != nil &&
            !manager.isScanning &&
            manager.isPowerOn &&
            manager.startScanWithMacs(targetMacs: targetMacs!))
        {
            rssiThreashHold = Int(self.tbRSSI.text!)
            btnStartStop.setTitle("Stop", for: .normal)
            tbRSSI.isUserInteractionEnabled = false
            tbTargetMacs.isUserInteractionEnabled = false
        }
    }
    
    func stopScanning() {
        manager.stopScan()
        btnStartStop.setTitle("Start", for: .normal)
        tbRSSI.isUserInteractionEnabled = true
        tbTargetMacs.isUserInteractionEnabled = true
    }
    
    @IBAction func touchUpStartButton(_ sender: Any) {
        if (manager.isScanning) {
            self.stopScanning()
            return
        }
        
        self.startScanning()
    }
}

